app/models/room.rb

```diff
@@ -1,3 +1,13 @@
 class Room < ApplicationRecord
    belongs_to :hotel
+   has_many :guest_rooms
+   has_many :guests, through: :guest_rooms
+
+   def hotel_name
+     hotel.name
+   end
+
+   def number_of_guests
+     guests.distinct.count
+   end
 end
```

Comment:
👍 

I don't mind the `hotel_name` method, but you don't necessarily need to make it a separate method either. The logic, `hotel.name`, isn't really making any inquiry to the database and is just pulling the name from the hotel object. So you could have just used that in the view.